### DISCUSSION TITLE
How to avoid useEffects from re-running when doing revalidateTag()?

### DISCUSSION THREAD

**Discussion by Hanskrogh** [vercel/next.js#65034] [App Router] [answered]
Hi,

I sit in the following situation, where we're building an ecommerce site for a customer. We have the following component to track "Product View Event in GTM"

```

const ProductEvents: FC<PropsWithChildren<ProductEventsProps>> = ({ product, selectedVariant }) => {
	useEffect(() => {
		if (product && selectedVariant) {
			viewItemEvent({
....
			});
		}
	}, []);

	return null;
};

ProductEvents.displayName = "ProductEvents";
export default React.memo(ProductEvents);


```

The problem is that everything you add/remove/update a variant to your cart - we call revalidateTag('cart'). But.... revalidateTag causes a rerender of the ENTIRE app, which causes the viewItemEvent to be triggered again. Do anyone have a solution to this problem?

**Comment by SidKH** [comment, marked as answer]
Hi 👋

React re-render does not remount the component.
The effect with the empty dependency list won't fire on every re-render.

https://github.com/vercel/next.js/assets/7762253/b4cd26d8-1feb-40e6-b418-dbb7bf090955

In your case, if the effect re-fires on every re-render (after revalidation) it could mean that you remount some components higher in the tree by updating the `key` or rendering components conditionally.

**Reply by Hanskrogh** [reply]
Thank you for the answer - I figured out that React.memo apperently caused my component to re-mount + re-render on revalidate.. :) But i solved the issue by removing React.memo wrapping the component :)
